### PR TITLE
WIP: feature: opportunity handler

### PR DESCRIPTION
**Pull request by gdenn**

### CODE REVIEW

**Comment by gdenn** [comment]
issue: #2

File: package.json [hunk 1]
```diff
@@ -49,9 +49,12 @@
     "@aws-lambda-powertools/logger": "^1.6.0",
     "@aws-lambda-powertools/metrics": "^1.6.0",
     "@aws-lambda-powertools/tracer": "^1.6.0",
+    "@aws-sdk/client-dynamodb": "^3.288.0",
     "@middy/core": "*",
     "@types/aws-lambda": "^8.10.111",
+    "@types/aws-sdk": "^2.7.0",
     "aws-cdk-lib": "^2.1.0",
+    "aws-sdk": "^2.1332.0",
```
Comment (dschro-1993) [inline package.json:57]: Will move current DynamoDB impl to V3 specs and later remove aws-sdk-v2

File: .projenrc.js [hunk 2]
```diff
@@ -12,6 +12,9 @@ const project = new awscdk.AwsCdkTypeScriptApp({
     '@aws-cdk/aws-lambda',
     '@aws-cdk/aws-appsync',
     '@aws-cdk/aws-dynamodb',
+    '@aws-sdk/client-dynamodb',
+    '@types/aws-sdk',
+    'aws-sdk',
```
Comment (dschro-1993) [inline .projenrc.js:17]: Will move current DynamoDB impl to V3 specs and remove aws-sdk-v2 after

File: .projenrc.js [hunk 3]
```diff
@@ -12,6 +12,9 @@ const project = new awscdk.AwsCdkTypeScriptApp({
     '@aws-cdk/aws-lambda',
     '@aws-cdk/aws-appsync',
     '@aws-cdk/aws-dynamodb',
+    '@aws-sdk/client-dynamodb',
+    '@types/aws-sdk',
```
Comment (dschro-1993) [inline .projenrc.js:16]: Please remove

Comment (dschro-1993) [inline .projenrc.js:16]:
```suggestion
```

File: src/graphql/lib/model.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+export interface Opportunity {
+  id: string;
+  name: string;
+  description: string;
+  reporter: User;
+  createdDate: Date;
+  lastModifiedDate: Date;
+}
+
+export interface CreateOpportunityRequest extends Opportunity { }
```
Comment (dschro-1993) [inline src/graphql/lib/model.ts:18]: Why do both Request-Objects extend Opportunity (with id and timestamps)?

File: src/graphql/lib/model.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
+export interface User {
+  id: string;
+  name: string;
+  email: string;
+  createdDate: Date;
+  lastModifiedDate: Date;
+}
+
+export interface Opportunity {
```
Comment (dschro-1993) [inline src/graphql/lib/model.ts:9]: Why does it not extend the common Entity Type in our libs?

File: src/graphql/lib/model.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
+export interface User {
```
Comment (dschro-1993) [inline src/graphql/lib/model.ts:1]: Why does it not extend the common Entity Type in our libs?

**Comment by dschro-1993** [comment]
Not happy with the Graphql schema, needs refinement.

**Comment by gdenn** [comment]
@dominik-schroettinger-rft updated the graphql schema and generate the interfaces with the graphql-code-gen. Can you have another look at the schema?

**Comment by gdenn** [comment]
continue review in #10

**Pull request closed without merging**